crates/worker/src/partition/state_machine/mod.rs

```diff
@@ -2821,13 +2878,17 @@ impl<S> StateMachineApplyContext<'_, S> {
                 info!(
                     "Will not run invocation {invocation_id} because it has been marked as completed/deleted already!"
                 );
-                VQueues::new(
-                    qid,
+                // we delete by id because we are not really sure if the invocation is still in
+                // Stage::Inbox or not.
+                VQueues::end_by_id(
                     self.storage,
                     self.vqueues_cache,
                     self.is_leader.then_some(self.action_collector),
+                    at,
+                    EntryKind::Invocation,
+                    invocation_id.partition_key(),
+                    &EntryId::from(invocation_id),
```

Comment:
This is a safety net in case that some other code paths that transitions the invocations status into `Completed` didn't clean up the vqueue entry, right?